bin/adhoc-scripts/updatePileupObjects.py

```diff
@@ -0,0 +1,185 @@
+                                 dest="url", default="https://cmsweb-testbed.cern.ch",
+                                 help="URL for the MSPileup service")
+        self.parser.add_argument("--cafile", required=False,
+                                 dest="cafile", help="CERN CA file")
+
+
+def getContext(client_cert, client_key, cafile):
+    """
+    Creates and return SSL context used by urllib module
+
+    :param client_cert: client's certificate pem file or proxy
+    :param client_key: client's key pem file or proxy
+    :param cafile: CERN CA certificate
+    :return: SSL context used by urllib
+    """
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+    context.load_cert_chain(certfile=client_cert, keyfile=client_key)
+    if cafile:
+        context.load_verify_locations(cafile=cafile)
+    return context
+
+def getPileupDocs(mspileupUrl, certDict, logger):
+    """
+    Fetch pileup documents from MSPileup
+    :param mspileupUrl: string with the MSPileup url
+    :param certDict: dictionary with cert and key reference
+    :param logger: logger object
+    :return: a list with the pileup documents
+    """
+    url = mspileupUrl + "/ms-pileup/data/pileup"
+    logger.debug("Fetching pileup documents from URL: %s, certs=%s", url, certDict)
+
+    # create context
+    context = getContext(certDict['cert'], certDict['key'], certDict['cafile'])
+
+    # place request
+    req = urllib.request.Request(url)
+    req.add_header('Accept', 'application/json')
+    resp = urllib.request.urlopen(req, context=context)
+
+    logger.debug("Response: %s status %s", resp, resp.status)
+    # check status ot HTTP response
+    if resp.status != 200:
+        msg = f"Bad response from server, status {resp.status}"
+        raise RuntimeError(msg)
+
+    # decode and load JSON data
+    sdata = resp.read().decode('utf8')
+    data = json.loads(sdata)
+
+    logger.debug("Response: %s", data)
+    if data and not data.get("result", []):
+        msg = f"Failed to retrieve pileup documents. Error: {data}"
+        raise RuntimeError(msg)
+    return data["result"]
+
+
+def writePileupDocs(mspileupUrl, puDocs, certDict, logger):
+    """
+    Update each pileup document in MSPileup
+    :param mspileupUrl: string with the MSPileup url
+    :param puDocs: list with pileup documents
+    :param certDict: dictionary with cert and key reference
+    :param logger: logger object
+    :return: None
+    """
+    url = mspileupUrl + "/ms-pileup/data/pileup"
+    headers = {'Content-Type': 'application/json'}
+    logger.info("Starting to update pileup documents against backend: %s", url)
+    cafile = certDict['cert']
+    capath = '/etc/grid-security/certificates'
+    for doc in puDocs:
+        logger.info("Updating pileup document: %s", doc.get("pileupName"))
+        req = urllib.request.Request(url)
+        req.add_header('Accept', 'application/json')
+        context = getContext(certDict['cert'], certDict['key'], certDict['cafile'])
```

Comment:
Perhaps a better approach would be to initialize the SSL context in the main function and simply pass it over to the read and write functions.

Reply:
sure, moved to main.